src/psyclone/tests/domain/gocean/transformations/gocean_opencl_trans_test.py

```diff
@@ -797,38 +780,62 @@ def test_opencl_options_effects():
 
     otrans = GOOpenCLTrans()
     otrans.apply(sched)
-    generated_code = str(psy.gen)
 
     # By default there is 1 queue, with an end barrier and local_size is 64
+    generated_code = str(psy.gen)
     assert "localsize = (/64, 1/)" in generated_code
     assert "ierr = clEnqueueNDRangeKernel(cmd_queues(1), " \
         "kernel_compute_cu_code, 2, C_NULL_PTR, C_LOC(globalsize), " \
         "C_LOC(localsize), 0, C_NULL_PTR, C_NULL_PTR)" in generated_code
-    assert "! Block until all kernels have finished\n" \
-        "      ierr = clFinish(cmd_queues(1))" in generated_code
+    assert "ierr = clFinish(cmd_queues(1))" in generated_code
     assert "ierr = clFinish(cmd_queues(2))" not in generated_code
 
+    # Reparse the example as changes are not possible after a psy.gen
```

Comment:
This is a temporary consequence of still mixing gen_code path and the backend. For this to work I need to call the lower_to_language_level in-tree during the generation and this cannot be reverted. When only the backend will be used the visitor entry point does a deep_copy of the full tree, so it can lower_to_languague_level without impacting the original/user tree.
When we finish porting the full OpenCL to use the backend this restriction can be lifted.